src/components/Slider/Slider.tsx

```diff
@@ -180,13 +180,14 @@ export class Slider extends BaseComponent<ISliderProps, ISliderState> implements
       currentValue = min + step * Math.round(currentSteps);
     }
 
+    if (onChange && currentValue !== this.state.value) {
```

Comment:
Minor tweak here; there are 2 concerns:
* You're now firing the change event before the state.value represents the new value. That means that for example, if the user were to read this.refs.slider.value within an onChange handler, it would still be the old value. I think we should be calling onChange when the value has been applied.
* setState is async, so you will need to pass in a callback that will be called once the state has been applied.

```tyepscript
let valueHasChanged = currentValue !== this.state.value;

this.setState({
  value: currentValue,
  renderedValue: renderedValue
}, () => {
  if (onChange && valueHasChanged) {
    onChange(this.state.value);
  }
}); 
```